utils/constants/MiscCodes.py

```diff
@@ -911,3 +911,33 @@ class ReputationSourceGain(IntEnum):
     REPUTATION_SOURCE_KILL = 0
     REPUTATION_SOURCE_QUEST = 1
     REPUTATION_SOURCE_SPELL = 2
+
+class ScriptTypes(IntEnum):
+    SCRIPT_TYPE_QUEST_START = 0
+    SCRIPT_TYPE_QUEST_END = 1
+    SCRIPT_TYPE_CREATURE_MOVEMENT = 2
+    SCRIPT_TYPE_CREATURE_SPELL = 3
+    SCRIPT_TYPE_GAMEOBJECT = 4
+    SCRIPT_TYPE_GENERIC = 5
+    SCRIPT_TYPE_GOSSIP = 6
+    SCRIPT_TYPE_SPELL = 7
+
+class TeleportToOptions(IntEnum):
+    TELE_TO_GM_MODE = 1
+    TELE_TO_NOT_LEAVE_TRANSPORT = 2
+    TELE_TO_NOT_LEAVE_COMBAT = 0x04
+    TELE_TO_NOT_UNSUMMON_PET = 0x08
+    TELE_TO_SPELL = 0x10
+    TELE_TO_FORCE_MAP_CHANGE = 0x20
+
+class CreatureAIEventTypes(IntEnum):
+    AI_EVENT_TYPE_OUT_OF_COMBAT = 1
+    AI_EVENT_TYPE_ON_LOW_HP = 2
```

Comment:
I think this should be renamed to `AI_EVENT_TYPE_HP` or similar. Seems to be used for events related to HP (not only low) according to VMaNGOS source code.
```cpp
    EVENT_T_HP                      = 2,                    // HPMax%, HPMin%, RepeatMin, RepeatMax
```

Reply:
Yeah I renamed it